### PR TITLE
JNG-4078 or else for all types

### PR DESCRIPTION
**Pull request by gprivi**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4078" title="JNG-4078" target="_blank">StoryJNG-4078</a>  orElse() for all types
  </td></table>
  <br />

### CODE REVIEW

File: judo-tatami-jsl-jsl2psm/src/test/resources/derived/TestDerivedExpressionModel.jsl [hunk 1]
```diff
@@ -34,6 +34,11 @@ entity Lead {
     field Integer value = 100000;
     field Boolean closed;
     relation SalesPerson salesPerson opposite leads;
+    derived Integer orElseValue => self.value!orElse(value = 10)!abs()!abs();
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2psm/src/test/resources/derived/TestDerivedExpressionModel.jsl:37]: is `!abs()!abs()` intentional?

File: judo-tatami-jsl-jsl2psm/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/derived/JslExpressionToJqlExpressionTest.java [hunk 2]
```diff
@@ -129,10 +129,14 @@ void testDerivedExpressionModel() throws Exception {
         assertEquals("\"String\nString2\"", jqlDerived("SalesPerson", "stringLiteral", "", ""));
         assertEquals("\"Raw\\n\"", jqlDerived("SalesPerson", "stringRawLiteral", "", ""));
         assertEquals("100.12", jqlDerived("SalesPerson", "decimalLiteral", "", ""));
-
         assertEquals("10 * 10 * 10", jqlDerived("SalesPerson", "powerOfTen", "", ""));
 
-    }
+        assertEquals("(self.value!isDefined() ? self.value!abs()!abs() : 10!abs()!abs())", jqlDerived("Lead", "orElseValue", "", ""));
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2psm/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/derived/JslExpressionToJqlExpressionTest.java:137]: is `!abs()!abs()` intentional?

**Review by bencegelei — APPROVED** [review]

File: judo-tatami-jsl-jsl2psm/src/test/resources/derived/TestDerivedExpressionModel.jsl (hunk 1 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/test/resources/derived/TestDerivedExpressionModel.jsl:37]: yes, it is intentional. It tests the chained function calls with no side effect.

File: judo-tatami-jsl-jsl2psm/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/derived/JslExpressionToJqlExpressionTest.java (hunk 2 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/derived/JslExpressionToJqlExpressionTest.java:137]: yes, it is intentional. It tests the chained function calls with no side effect.

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**